content/courses/native-onchain-development/program-derived-addresses.md

```diff
@@ -631,16 +621,17 @@ let pda_counter = next_account_info(account_info_iter)?;
 let system_program = next_account_info(account_info_iter)?;
 ```
 
-Next, there's a check to make sure `total_len` is less than 1000 bytes, but
-`total_len` is no longer accurate since we added the discriminator. Let's
-replace `total_len` with a call to `MovieAccountState::get_account_size`:
+Next, Calculate the account size dynamically, accounting for all data fields and potential growth.
+Define maximum sizes using constants to avoid magic numbers:
 
 ```rust
-let account_len: usize = 1000;
+const MAX_TITLE_LENGTH: usize = 256;
+const MAX_DESCRIPTION_LENGTH: usize = 1024;
 
-if MovieAccountState::get_account_size(title.clone(), description.clone()) > account_len {
-    msg!("Data length is larger than 1000 bytes");
-    return Err(ReviewError::InvalidDataLength.into());
+impl MovieAccountState {
+    pub fn get_account_size(title: &str, description: &str) -> usize {
+        8 + 1 + (4 + title.len().min(MAX_TITLE_LENGTH)) + (4 + description.len().min(MAX_DESCRIPTION_LENGTH))
+    }
```

Comment:
We should use INIT_SPACE for this. https://www.anchor-lang.com/docs/space (scroll down to 'The InitSpace macro
').